packages/fxa-shared/subscriptions/stripe.ts

```diff
@@ -189,3 +191,43 @@ export function singlePlan(
   const subItems = subscription.items.data;
   return subItems.length > 1 ? null : subItems[0].plan;
 }
+
+/**
+ * Given two plans, A and B, determine whether B is eligible for a subscription
+ * update (upgrade/downgrade) from A.
+ */
+export const getSubscriptionUpdateEligibility: (
+  x: Plan,
+  y: Plan
+) => SubscriptionUpdateEligibility = (currentPlan: Plan, newPlan: Plan) => {
+  const currentPlanMetaData = metadataFromPlan(currentPlan);
+  const newPlanMetaData = metadataFromPlan(newPlan);
+  const currentOrder =
+    !!currentPlanMetaData.productOrder &&
+    parseInt(currentPlanMetaData.productOrder);
+  const newOrder =
+    !!newPlanMetaData.productOrder && parseInt(newPlanMetaData.productOrder);
+
+  if (
+    currentPlan.plan_id === newPlan.plan_id ||
+    !currentPlanMetaData.productSet ||
+    currentPlanMetaData.productSet !== newPlanMetaData.productSet ||
+    !currentOrder ||
+    Number.isNaN(currentOrder) ||
+    !newOrder ||
+    Number.isNaN(newOrder) ||
+    newOrder === currentOrder
+  ) {
+    return SubscriptionUpdateEligibility.INVALID;
+  }
```

Comment:
> 
> 
> It seems like a couple of these are not necessary conditions to check, as they are implicit from other checks:
> 
>     * `newOrder === currentOrder` is true if neither of the other two `if` blocks apply.
> 

That's true.  I'm a fan of early returns, but I guess in this case it doesn't really save us much.  What about readability?  Is it better to have the code to explicitly show that `newOrder === currentOrder` is invalid or implicit via the default return at the end?  Or makes no difference either way?

Your point is valid and I don't have strong feelings about this.  I'll just leave it unless it really bugs you.

>     * `currentPlan.plan_id === newPlan.plan_id` is implicitly checked when both `productOrder` and `productSet` are the same.

I see what you are getting it, but it's possible that the plan ids don't match while the `productOrder` and `productSet` are the same since anything goes when it comes to Stripe metadata.  The return value would still be correct without that comparison, which I think is your point here.